rs/registry/canister/tests/data_migration_test.rs

```diff
@@ -0,0 +1,325 @@
+impl DataMigrationAssert for EmptyDataAssertion {
+    async fn assert_expected_changes(&self, new_mutations: &[RegistryMutation]) {
+        assert!(new_mutations.is_empty())
+    }
+}
+
+fn mutations_to_json(mutations: &[RegistryMutation]) -> String {
+    let mapped: BTreeMap<String, Value> = mutations
+        .iter()
+        .map(|m| {
+            let key = m.key_as_string();
+            let value = if m.mutation_type() == Type::Delete {
+                Value::String("Deleted".to_string())
+            } else if key.starts_with(NODE_RECORD_KEY_PREFIX) {
+                let decoded = NodeRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with("subnet_list") {
+                let decoded = SubnetListRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(NODE_REWARDS_TABLE_KEY) {
+                let decoded = NodeRewardsTable::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with("blessed_replica_versions") {
+                let decoded = BlessedReplicaVersions::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(API_BOUNDARY_NODE_RECORD_KEY_PREFIX) {
+                let decoded = ApiBoundaryNodeRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(NODE_OPERATOR_RECORD_KEY_PREFIX) {
+                let decoded = NodeOperatorRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(REPLICA_VERSION_KEY_PREFIX) {
+                let decoded = ReplicaVersionRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(HOSTOS_VERSION_KEY_PREFIX) {
+                let decoded = HostosVersionRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(SUBNET_RECORD_KEY_PREFIX) {
+                let decoded = SubnetRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else if key.starts_with(DATA_CENTER_KEY_PREFIX) {
+                let decoded = DataCenterRecord::decode(m.value.as_slice()).unwrap();
+                serde_json::to_value(&decoded).unwrap()
+            } else {
+                panic!("Missing mapping for key {}", key);
+            };
+
+            let value = fixup_ids(value);
+
+            (key.to_string(), value)
+        })
+        .collect();
+
+    serde_json::to_string_pretty(&mapped).unwrap()
+}
+
+fn fixup_ids(mut value: Value) -> Value {
+    if let Value::Array(arr) = &value {
```

Comment:
"arr" is not a real word; "array" is though. Stick to real words. In this case, it perfectly matches with `Value::Array`, so there's _another_ strong reason to go with `array`.